source/base/parameter_handler.cc

```diff
@@ -2311,13 +2319,391 @@ ParameterHandler::print_parameters (std::ostream     &out,
+           p = current_section.begin();
+           p != current_section.end(); ++p)
+        if (is_parameter_node (p->second) == true)
+          longest_name = std::max (longest_name,
+                                   demangle(p->first).length());
+
+      // print entries one by one. make sure they are sorted by using
+      // the appropriate iterators
+      for (boost::property_tree::ptree::const_assoc_iterator
+           p = current_section.ordered_begin();
+           p != current_section.not_found(); ++p)
+        if (is_parameter_node (p->second) == true)
+          {
+            // print name and value
+            out << std::setw(overall_indent_level*2) << ""
+                << "set "
+                << demangle(p->first)
+                << std::setw(longest_name-demangle(p->first).length()+1) << " "
+                << " = ";
+
+            // print possible values:
+            const unsigned int pattern_index = p->second.get<unsigned int> ("pattern");
+            const std::string full_desc_str = patterns[pattern_index]->description (Patterns::PatternBase::Text);
+            const std::vector<std::string> description_str
+              = Utilities::break_text_into_lines (full_desc_str,
+                                                  78 - overall_indent_level*2 - 2, '|');
+            if (description_str.size() > 1)
+              {
+                out << std::endl;
+                for (unsigned int i=0; i<description_str.size(); ++i)
+                  out << std::setw(overall_indent_level*2+6) << ""
+                      << description_str[i] << std::endl;
+              }
+            else if (description_str.empty() == false)
+              out << "  " << description_str[0] << std::endl;
+            else
+              out << std::endl;
+
+            // if there is a documenting string, print it as well
+            if (p->second.get<std::string>("documentation").length() != 0)
+              out << std::setw(overall_indent_level*2 + longest_name + 10) << ""
+                  << "(" << p->second.get<std::string>("documentation") << ")" << std::endl;
+          }
+
+      break;
+    }
+
+    default:
+      Assert (false, ExcNotImplemented());
+    }
+
+
+  // if there was text before and there are sections to come, put two
+  // newlines between the last entry and the first subsection
+  {
+    unsigned int n_parameters = 0;
+    unsigned int n_sections   = 0;
+    for (boost::property_tree::ptree::const_iterator
+         p = current_section.begin();
+         p != current_section.end(); ++p)
+      if (is_parameter_node (p->second) == true)
+        ++n_parameters;
+      else if (is_alias_node (p->second) == false)
+        ++n_sections;
+
+    if ((style != Description)
+        &&
+        (!(style & 128))
+        &&
+        (n_parameters != 0)
+        &&
+        (n_sections != 0))
+      out << std::endl << std::endl;
+  }
+
+  // now traverse subsections tree, in alphabetical order
```

Comment:
Yes -- we use `current_section.ordered_begin()` to get an iterator that uses alphabetical order.